codice/ddf
# DDF-4043 As a user I would like to be able to export a result set as a HTML file so that I can have a html file to use externally.

This PR is dependent on [DDF-3997](https://github.com/codice/ddf/pull/3516) to export a result set. Single metacards do however work. 

Some additional styling is on it's way so for the moment it is fairly basic. This will be included before this PR is merged. 

#### What does this PR do?

**Description**
There are two transformers one for a single metacard and one for exporting a result set. Each metacard is exported to HTML and is structured the same way that a metacard is structured with categories and attributes for organization. 

**Acceptance Criteria**
Support generating an HTML file representing the result list. This should include a detailed view of each record's elements, and image references pointing at thumbnails. 

**Food for Thought**
What other attributes should be included? How do you like the structure of the metacard export? There is an existing metacard to html bundle however some of its components are non-functional and it does not provide a detailed view. 

#### Who is reviewing it? 
@jrnorth 
@leo-sakh 
@Bdthomson 

#### Select relevant component teams: 
https://github.com/orgs/codice/teams
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@coyotesqrl 


#### How should this be tested?

1. Verify that the bundle has started.
2. Create a metacard and then export that single metacard as HTML.
3. Verify that it contains its attributes and that the thumbnail exists (if the metacard contained one). 

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-3997](https://codice.atlassian.net/browse/DDF-3997)
#### Screenshots/Example Output

Here is an example output. This metacard contained very little data and a thumbnail. 
[index.htm.zip](https://github.com/codice/ddf/files/2271431/index.htm.zip)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
